### PR TITLE
Feature/config filepath

### PR DESCRIPTION
**Pull request by shreyas-sriram**
Addresses https://github.com/N0MoreSecr3ts/wraith/issues/94.

This was a tricky one. I needed access to the CLI args before calling `SetConfig()`, however this is not possible because `init()` runs before everything. The most common way to setup a custom configuration file was to do it in the callback function to `cobra.OnInitialize()`.

I used global `viper` instance to bind the flags and eventually call `SetConfig()`, which reads in the config and initializes `WraithConfig`. `WraithConfig` was also moved to `core` package to prevent import cycles **(cmd -> core -> cmd)**. Following this, some function definitions were updated.

***Important: The CLI values overrides the values read from the config file.***

### CODE REVIEW

**Comment by mattyjones** [comment]
@shreyas-sriram So that I am clear the order of config is
1. CLI
2. Config File
3. Defaults

What happens when I put certain values on the commandline and others I pull from the config and the remainder from the defaults? Ex. Orgs and repos on the commandline, scan settings in the config file, and then use defaults for the url, etc.

You should be able to mix and match given the above order of operations.

File: cmd/scanGithub.go [hunk 1]
```diff
@@ -109,12 +110,12 @@ func init() {
...
-	err = wraithConfig.BindPFlag("github-orgs", scanGithubCmd.Flags().Lookup("github-orgs"))
-	err = wraithConfig.BindPFlag("github-repos", scanGithubCmd.Flags().Lookup("github-repos"))
-	err = wraithConfig.BindPFlag("github-users", scanGithubCmd.Flags().Lookup("github-users"))
-	err = wraithConfig.BindPFlag("commit-depth", scanGithubCmd.Flags().Lookup("commit-depth"))
+	err := viper.BindPFlag("add-org-members", scanGithubCmd.Flags().Lookup("add-org-members"))
+	err = viper.BindPFlag("github-api-token", scanGithubCmd.Flags().Lookup("github-api-token"))
+	err = viper.BindPFlag("github-orgs", scanGithubCmd.Flags().Lookup("github-orgs"))
+	err = viper.BindPFlag("github-repos", scanGithubCmd.Flags().Lookup("github-repos"))
+	err = viper.BindPFlag("github-users", scanGithubCmd.Flags().Lookup("github-users"))
+	err = viper.BindPFlag("commit-depth", scanGithubCmd.Flags().Lookup("commit-depth"))
```
Comment (mattyjones) [inline cmd/scanGithub.go:118]: @shreyas-sriram These should be alpha sorted to make things easy to find.

File: cmd/scanGitlab.go [hunk 2]
```diff
@@ -65,10 +66,10 @@ func init() {
 	scanGitlabCmd.Flags().StringSlice("gitlab-projects", nil, "List of Gitlab projects or users to scan")
 	scanGitlabCmd.Flags().Float64("commit-depth", -1, "Set the commit depth to scan")
 
-	err := wraithConfig.BindPFlag("commit-depth", scanGitlabCmd.Flags().Lookup("commit-depth"))
-	err = wraithConfig.BindPFlag("add-org-members", scanGitlabCmd.Flags().Lookup("add-org-members"))
-	err = wraithConfig.BindPFlag("gitlab-api-token", scanGitlabCmd.Flags().Lookup("gitlab-api-token"))
-	err = wraithConfig.BindPFlag("gitlab-projects", scanGitlabCmd.Flags().Lookup("gitlab-projects"))
+	err := viper.BindPFlag("commit-depth", scanGitlabCmd.Flags().Lookup("commit-depth"))
```
Comment (mattyjones) [inline cmd/scanGitlab.go:69]: @shreyas-sriram These should be alpha sorted.

**Comment by shreyas-sriram** [comment]
From my testing, the current ***order of priority*** would be

1. CLI
2. Config file
3. Defaults

I am unaware of specific edge cases to test. Do you have any?

**Review by mattyjones — APPROVED** [review]
